### DISCUSSION TITLE
Database migration to postgre SQL instead of MongoDB

### DISCUSSION THREAD

**Discussion by bekogeko** [Pulth-Team/Pulth#9] [Ideas]
Currently, Prisma for MongoDB has some problems with relational fields and cycles. The main problem with MongoDB and Prisma is in the comment system there is a self-relation in a comment model.

There is an ideal model for the comment.
```prisma
model Comment{
    parent   Comment?  @relation("ParentChild", fields: [parentId], references: [id], onUpdate: Cascade, onDelete: Cascade)
    children Comment[] @relation("ParentChild")
    parentId String?   @db.ObjectId
}
```
Take a look at the `Cascade` value in the `onUpdate` and `onDelete` values. The `Cascade` basically means **_do the same thing for me_**. For example, if we delete the parent, we will delete this comment too, and the same thing applies to `onUpdate` too. If the parent changes its id then update my  `parentId` too.

But eventually, this cannot be achieved that easily because Prisma for MongoDB doesn't support self-relations except relation's `onUpdate` and `onDelete` actions set to `NoAction`. I think it's a deal-breaker for us but here I am for the discussions.

### What if we switch to a fully supported Prisma connector?
That way we can freely use the full potential of the prisma while enjoying its beautiful schema and ORM system. But must keep in mind that this approach requires a lot of work and a lot of changes in the Data layer which can break most things.

Few examples of the problems:
- [relational cycle problems ](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions/special-rules-for-referential-actions#cyclic-relation-between-three-tables-sql-server-and-mongodb) (self relations)
- [The Differences to relational databases](https://www.prisma.io/docs/concepts/database-connectors/mongodb#differences-to-connectors-for-relational-databases)

**Comment by dhincal** [comment]
In my opinion, we don't need to use the `Cascade` value in `onUpdate` because we don't really need to update the child comments when a parent updates (if we don't change `ID` of the parent and I think we shouldn't). But if we need to change the child when parent changes, then it looks like there is no option but switch the Prisma Connector.

**Comment by bekogeko** [comment]
@dhincal 

I have another solution that doesn't require changing the MongoDB Provider. When we delete, we can delete the content and the shown authorName.
 
here is an example comment tree. (the id's of the comments were indicated as #.id)
```
- Bekogeko #.1
  a bad comment which will be deleted.
     - Deniz Hincal #.2
        so rude please delete this comment.
- Deniz Hincal #.3
   another root comment 
```

after we delete the #.1 comment the comment tree will look like this.
```
- comment deleted
  comment deleted 
     - Deniz Hincal #.2
        so rude please delete this comment.
- Deniz Hincal #.3
   another root comment 
```

with that approach, we can also delete comments on account deletion. Here is another tree for that
````
- account deleted
  comment deleted 
   - Deniz Hincal #.2
        so rude please delete this comment.
- Deniz Hincal #.3
   another root comment 
```

**Comment by bekogeko** [comment]
A solution https://github.com/prisma/prisma/discussions/17746
